### PR TITLE
[Test] Add %host_triple and %host_sdk substitutions

### PR DESCRIPTION
**Pull request by rintaro**
Macro tests need to build host libraries/tools. We can't use `%target-*` substitutions for that.

rdar://107398734

### CODE REVIEW

**Comment by rintaro** [comment]
@swift Please smoke test

File: test/Macros/composed_macro.swift [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // RUN: %empty-directory(%t)
-// RUN: %target-build-swift -swift-version 5 -I %swift-host-lib-dir -L %swift-host-lib-dir -emit-library -o %t/%target-library-name(MacroDefinition) -module-name=MacroDefinition %S/Inputs/syntax_macro_definitions.swift -g -no-toolchain-stdlib-rpath
+// RUN: %swiftc_driver -sdk %host_sdk -target %host_triple -swift-version 5 -I %swift-host-lib-dir -L %swift-host-lib-dir -emit-library -o %t/%target-library-name(MacroDefinition) -module-name=MacroDefinition %S/Inputs/syntax_macro_definitions.swift -g -no-toolchain-stdlib-rpath
```
Comment (bnbarham) [inline test/Macros/composed_macro.swift:2]: Could we have a `host-build-swift` to match `target-build-swift`?

File: test/lit.cfg [hunk 2]
```diff
@@ -963,7 +963,7 @@ def use_interpreter_for_simple_runs():
         result += (
             'env SWIFT_INTERPRETER=%r %s %r %s -module-name main %s %s %s '
             % (config.swift, xcrun_prefix, config.swift, target_options,
-               config.swift_test_options,
+               onfig.swift_test_options,
```
Comment (bnbarham) [inline test/lit.cfg:966]: Whoops!

**Review by bnbarham — COMMENTED** [review]
Thanks Rintaro!

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test macOS

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: test/Macros/composed_macro.swift [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // RUN: %empty-directory(%t)
-// RUN: %target-build-swift -swift-version 5 -I %swift-host-lib-dir -L %swift-host-lib-dir -emit-library -o %t/%target-library-name(MacroDefinition) -module-name=MacroDefinition %S/Inputs/syntax_macro_definitions.swift -g -no-toolchain-stdlib-rpath
+// RUN: %host-build-swift -swift-version 5 -I %swift-host-lib-dir -L %swift-host-lib-dir -emit-library -o %t/%target-library-name(MacroDefinition) -module-name=MacroDefinition %S/Inputs/syntax_macro_definitions.swift -g -no-toolchain-stdlib-rpath
```
Comment (bnbarham) [inline test/Macros/composed_macro.swift:2]: Any thoughts on including `-I %swift-host-lib-dir -L %swift-host-lib-dir` by default for `host-build-swift`? Or we could have `%host-tool-build-swift` or something 😅. What you have now is fine though.

**Review by bnbarham — APPROVED** [review]

**Pull request merged**